tests/test_RecursiveFetaureElimination.py

```diff
@@ -1,6 +1,98 @@
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.datasets import make_friedman1
+from sklearn.linear_model import LinearRegression
+
 from feature_selection import recursive_feature_elimination
 
 
-def test_1():
-    result = recursive_feature_elimination(1, 2, 3, 4, 5)
-    assert result == []
+def scorer(X, y):
+    """
+    Sample custom scorer that fits a model and returns
+    an appropriate score for the feature selection problem.
+
+    Parameters
+    ----------
+    X : array-like of shape (n_samples, n_features)
+        Test samples
+
+    y : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        True values for X, used for training
+
+    Returns
+    -------
+    Index of feature column with the lowest weight.
+    """
+    model = LinearRegression()
+    model.fit(X, y)
+    return X.columns[model.coef_.argmin()]
+
+
+def test_recursive_feature_elimination():
+    """
+    This test creates a dataset that has 5 features that are are used to compute `y`.
+    The remaining 5 features are independent of `y`.
+    This test should select the 5 feature columns used to compute `y`.
+    """
+    X, y = make_friedman1(n_samples=200, n_features=10, random_state=10)
+    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=5)
+    assert features == [0, 1, 3, 4, 9]
+
+    # Test with n_features_to_select something other than 0.5 number of total features
+    # to ensure logic to stop feature elimination works correctly
+    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=3)
+    assert len(features) == 3
+
+    # Retest with column names
+    X, y = make_friedman1(n_samples=200, n_features=10, random_state=10)
+    X = pd.DataFrame(X, columns=['zero', 'one', 'two', 'three', 'four',
+                                 'five', 'six', 'seven', 'eight', 'nine'])
+    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=5)
+    assert features == ['zero', 'one', 'three', 'four', 'nine']
```

Comment:
Same comment as before.

```suggestion
    features = recursive_feature_elimination(scorer, X, y, n_features_to_select=4)
    assert features == ['zero', 'one', 'three', 'four']
```